internal/semver/semver.go

```diff
@@ -115,3 +218,20 @@ func (v *Semver) UnmarshalJSON(data []byte) error {
 	*v = version
 	return nil
 }
+
+// ByVersion implements [sort.Interface] for sorting semantic version strings.
+// Copied from Go's semver pkg with minimal modification.
+// https://cs.opensource.google/go/x/mod/+/master:semver/semver.go
+type ByVersion []Semver
```

Comment:
nit: `SortByVersion` to indicate sorting functionality?

Reply:
I made the type private so it's not used outside of semver.Sort.
When using this in sort.Sort the call would look like `sort.Sort(SortByVersion(foo))`. I think `sort.Sort(ByVersion(foo))` is more elegant?